src/databricks/labs/ucx/assessment/crawlers.py

```diff
@@ -24,6 +23,8 @@
 ]
 
 _AZURE_SP_CONF_FAILURE_MSG = "Uses azure service principal credentials config in "
+_SECRET_PATTERN = r"{{(secrets.*?)}}"
+_AZURE_SPN_LIST = []
```

Comment:
Don't store global mutable state. Call API for the second time in crawler for service principals